components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```diff
@@ -348,6 +350,46 @@ public static Tenant initializeTenant(TenantInfoBean tenantInfoBean) {
         return tenant;
     }
 
+    public static void updateTenantPassword(TenantInfoBean tenantInfoBean, String adminPassword) throws Exception {
+
+        if (StringUtils.isBlank(adminPassword)) {
+            return;
+        }
+        UserStoreManager userStoreManager;
+        try {
+            userStoreManager = getUserStoreManager(tenantInfoBean.getTenantId());
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            String msg = "Error in getting the user store manager for tenant, tenant domain: " +
+                    tenantInfoBean.getTenantDomain() + ".";
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new Exception(msg, e);
```

Comment:
Shall we throw specific exception type instead of generic ?

Reply:
Fixed in https://github.com/wso2/carbon-multitenancy/pull/276/commits/3fd11ef5ee2c9ee187a4ec951bb8b5eb59acd6e1